### PR TITLE
fix: set to empty content when removing all

### PR DESCRIPTION
**Pull request by TomasSlama**
[FX-2505]

### Description

When we select all text and want to remove it with DEL or Backspace, there is an edge case, where we don’t remove everything.

### How to test

1. Click ordered/unordered list button
2. Type in hello world
3. Press Enter
4. Press CTRL + A
5. Press DELETE / Backspace

**expected:**
- selected text and formatting are deleted

**previous behaviour**
- `<p><ol><li><br></li></ol></p>` was value of editor

### Review

- [ ] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [ ] Annotate all `props` in component with documentation
- [ ] Create `examples` for component
- [ ] Ensure that deployed demo has expected results and good examples
- [ ] Ensure that tests pass by running `yarn test`
- [ ] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [ ] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-2505]: https://toptal-core.atlassian.net/browse/FX-2505?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by OndrejTuma — DISMISSED** [review]

File: packages/picasso/src/QuillEditor/hooks/useQuillInstance/useQuillInstance.tsx [hunk 1]
```diff
@@ -14,28 +15,31 @@ export type EditorOptionsType = {
...
-      this.quill.format(key, false, Quill.sources.USER)
-    })
-  }
+  const isUserSelectionNonZero = !collapsed
+  const textLength = this.quill.getLength() - 1
+  const selection = this.quill.getSelection()
+  const isSelectedEverything = textLength === selection?.length
 
-  return true
+  if (isUserSelectionNonZero && isSelectedEverything) {
```
[the remark this answers is not in the document]
Comment (TomasSlama) [reply packages/picasso/src/QuillEditor/hooks/useQuillInstance/useQuillInstance.tsx:37]: it makes sense, you are right

**Comment by TomasSlama** [comment]
@LashaJini thanks for pointing out, that the previous solution didn't work correctly for cutting text out with `CTRL + X`. So I have come up with a better solution.

**Comment by LashaJini** [comment]
CTRL-X + CTRL-V

https://user-images.githubusercontent.com/72510037/154918836-38fad937-0f8f-4178-9d15-cc2ebd77cc3f.mp4

**Comment by LashaJini** [comment]
CTRL-X + CTRL-V

https://user-images.githubusercontent.com/72510037/154919500-9ef98983-fb19-459d-8441-da9be36d5932.mp4

File: packages/picasso/src/QuillEditor/hooks/useSubscribeToQuillEvents/useSubscribeToQuillEvents.tsx [hunk 2]
```diff
@@ -45,21 +46,37 @@ const useSubscribeToQuillEvents = ({
...
+  }, [quill])
+
   useEffect(() => {
     if (!quill) {
       return
     }
 
     quill.on('selection-change', selectionChangeHandler)
     quill.on('text-change', textChangeHandler)
+    quill.on('text-change', cleanupOnAllContentRemoval)
```
Comment (denieler) [inline packages/picasso/src/QuillEditor/hooks/useSubscribeToQuillEvents/useSubscribeToQuillEvents.tsx:64]: all the rest callbacks happen in this file have `*Handler` postfix, it seems adding `cleanupOnAllContentRemoval` is not following this component design principles

**Review by OndrejTuma — APPROVED** [review]
Looks good to me. Agree with @denieler that name should be updated. Otherwise 👍

**Review by dmaklygin — APPROVED** [review]

**Review by TomasSlama — COMMENTED** [review]
LGTM 👍

**Review by augustobmoura — APPROVED** [review]
